src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,291 @@
+import edu.kit.ipd.crowdcontrol.objectservice.mail.MailHandler;
+import edu.kit.ipd.crowdcontrol.objectservice.template.Template;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Result;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Describes a Manager for money transfers. The Manager can log payments and pay off.
+ *
+ * @author Felix Rittler
+ */
+public class MoneyTransferManager {
+
+    private static final Logger LOGGER = LogManager.getLogger(MoneyTransferManager.class);
+    private MailHandler mailHandler;
+    private WorkerBalanceOperations workerBalanceOperations;
+    private WorkerOperations workerOperations;
+    private int payOffThreshold;
+    private String notificationMailAddress;
+    private StringBuilder notificationText;
+    private ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> schedule = null;
+
+    /**
+     * Creates a new instance of the MoneyTransferManager
+     *
+     * @param mailHandler             the mailhandler, used to fetch new giftcodes and send notification and payment messages
+     * @param workerBalanceOperations the workerBalanceOperations, used to change the credit balance of a worker
+     * @param workerOperations        the workerOperations, used to do operations on workers
+     * @param notificationMailAddress the mail address to send notifications
+     */
+    public MoneyTransferManager(MailHandler mailHandler, WorkerBalanceOperations workerBalanceOperations, WorkerOperations workerOperations, String notificationMailAddress) {
+        this.mailHandler = mailHandler;
+        this.workerOperations = workerOperations;
+        this.workerBalanceOperations = workerBalanceOperations;
+        this.payOffThreshold = 0;
+        this.notificationMailAddress = notificationMailAddress;
+        this.notificationText = new StringBuilder();
+    }
+
+    /**
+     * Starts the MoneyTransferManager, so giftcodes become submitted to workers every 7 days.
+     */
+    public synchronized void start() {
+        if (schedule != null) {
+            throw new IllegalStateException("run() was called twice!");
+        }
+
+        Runnable runnable = () -> {
+            try {
+                submitGiftCodes();
+            } catch (MoneyTransferException e) {
+                try {
+                    sendNotification(e.toString(), true);
+                } catch (MoneyTransferException f) {
+                    LOGGER.error(f);
+                }
+                LOGGER.error(e);
+            }
+        };
+
+
+        schedule = scheduler.scheduleAtFixedRate(runnable, 7, 7, TimeUnit.DAYS);
```

Comment:
`7` should probably a constant in the class, so it's easy to find and change. Something like `SCHEDULE_INTERVAL_DAYS`